tests-performance/README.md

```diff
@@ -0,0 +1,52 @@
+### Profiling and Performance Testing of `libmata`
+
+#### Structure
+
+1. `automata/`: short database of approximately 70 automata of various complexity. The automata come from different
+  benchmarks. `easiest` benchmarks should take less then second, while, `hardest` can take over 60 seconds.
+
+2. `bin/`: target folder for compiling binaries used for profiling or performance testing.
+3. `src/`: source codes that will be compiled to `bin` directory. The directory contains several templates:
+  * `template.cc` contains basic template for arbitrary code to be profiled or performance tested.
+  * `template-with-cli.args.cc` contains basic template that takes file as an command line argument, parses the automaton
+     and computes minterms.
+  * `template-with-list-of-automata.cc` contains basic template that without arguments takes a hardcoded list of input 
+
+     automata taken from `automata` directory; in case it is run with arguments, it reads the input files that contains
+     paths to source automata. 
+4. `pycobench`: a performance testing script
+5. `pyco_proc`: a parser of performance results
+6. `scripts/`: several helper scripts for profiling
+* `scripts/run_callgrind.sh` to print top functions (exclusive time, greater than 1%)
+* `scripts/run_massif.sh` to print memory peak in [B]
+
+#### How to profile
+
+1. Copy one of the `template*.cc` files with boilerplate code.
+2. Find comment `HERE COMES YOUR CODE THAT YOU WANT TO PROFILE` and fill your code.
+3. Run `make` from `src` or compile your binary.
+4. Run `callgrind` or custom profile on the binary.
+5. You can use one of the scripts:
+  * `scripts/run_callgrind.sh` to print top functions (exclusive time, greater than 1%)
+  * `scripts/run_massif.sh` to print memory peak in [B]
+
+#### How to test performance
+
+1. Copy one of the `template*.cc` files with boilerplate code.
+2. Find comment `HERE COMES YOUR CODE THAT YOU WANT TO PROFILE` and fill your code.
+3. Run `make` from `src` or compile your binary.
+4. Add the new binary to the `jobs.yaml` as follows:
+5. You can use one of the scripts:
+* `scripts/run_callgrind.sh` to print top functions (exclusive time, greater than 1%)
+* `scripts/run_massif.sh` to print memory peak in [B]
```

Comment:
This should probably not be here

Reply:
Why not? Can you elaborate?

Reply:
It looks like it is copy-pasted from previous section and it does not fit there (there are two steps number 5, and the following part directly connects to number 4.

Reply:
I reworded it.

Reply:
Did you really fix it? I still see the code block for `jobs.yaml` below the step 5.

Reply:
I think you changed something under `How to profile` which was all right, this stuff is under `How to test performance`.

Reply:
Now really resolved. It is mentioned exhaustively in `scripts` part only.